util/progress/progressui/display.go

```diff
@@ -1123,3 +1260,78 @@ func wrapHeight(j []*job, limit int) []*job {
 	}
 	return wrapped
 }
+
+func byteArrayToStringArray(in [][]byte) []string {
+	if len(in) == 0 {
+		return nil
+	}
+
+	out := make([]string, len(in))
+	for i, b := range in {
+		out[i] = string(b)
+	}
+	return out
+}
+
+type logMux struct {
+	byDigest map[digest.Digest]*digestLogs
+}
+
+func newLogMux() *logMux {
+	return &logMux{
+		byDigest: make(map[digest.Digest]*digestLogs),
+	}
+}
+
+func (mux *logMux) update(log *client.VertexLog) {
+	logs, ok := mux.byDigest[log.Vertex]
+	if !ok {
+		logs = &digestLogs{
+			lastByStream: make(map[int]*logEntry),
+		}
+		mux.byDigest[log.Vertex] = logs
+	}
+	logs.update(log)
+}
+
+type digestLogs struct {
+	entries      []*logEntry
```

Comment:
Wouldn't this be easier with just one field with type `map[int][]*logEntry`?

Reply:
I'm trying to remember why I did it this way and I'm drawing a blank so I'll change this to use a map.